body.tex

```diff
@@ -1,6 +1,136 @@
+        \bottomrule
+    \end{tabular}
+\end{table}
+
+Managing a database of this size is a challenge. In 2023, we moved from InfluxDB OSS to InfluxDB Enterprise cluster to improve the database scalability and to have access to incremental backup and restore tools, available in the Enterprise product only, but essential for managing the databases in Sasquatch.
+
+At the USDF environment, the InfluxDB Enterprise cluster currently has 2x8-core data nodes with 30\,TB of local SSD storage each and a replication factor (RF) of two for redundancy. As the Sasquatch databases grow in size, we plan on adding more data nodes to the InfluxDB Enterprise cluster to increase the storage capacity while keeping a RF of two. We are actively working with InfluxData support to understand the best practices for scaling the InfluxDB cluster and leveraging the new features available in InfluxDB 3.0, such as storage tiering and the fact that the new version is Kubernetes native.
+
+We use MirrorMaker2 to mirror the data from the Summit to the USDF. With a retention period of 72h in Kafka at the Summit, MirrorMaker2 can automatically recover from network or power outages at USDF. While less frequent, we can recover from outages longer than 72h by restoring data from Summit backups to the USDF InfluxDB instance, however this process still needs automation. In situations where the data is in Kafka but missing in the InfluxDB either at the Summit or USDF environments we can deploy repairer connectors to replay the missing data to InfluxDB. We are also working on improving data consistency by implementing data validation checks in the databases at the Summit ant USDF.
+
+In addition to mirroring data from the Summit to USDF, we also need to support sending metrics computed at USDF back to the Summit in real-time. This requires running another instance of MirrorMaker2 at the Summit to be able to mirror data in both directions. We are currently working on making this feature more reliable in Sasquatch with respect to managing schemas in both environments.
+
+At the Summit environment, the OCS is adopting Apache Kafka for its middleware implementation \cite{2024SPIE13101.59Ftmp, TSTN-033}, which means that the OCS will record commands, events and telemetry data in Kafka in the first place. That puts Apache Kafka into a central role at the Summit, and in principle, we can use the same Kafka cluster for Sasquatch. This change  would simplify the integration of the OCS and Sasquatch at the Summit. For example, the Kafka producers currently used to send data to Kafka won't be necessary, eliminating the data duplication between the OCS and Sasquatch. However, adopting the same Kafka cluster for both the OCS and Sasquatch also brings new challenges. We must configure the OCS Kafka producers properly to achieve the low latency requirements for commands and events topics while prioritizing data durability for telemetry topics.
```

Comment:
maybe "in Kafka in the first place" -> "directly into Kafka".  Then there's some verb tense stuff.  I think this all ought to be future.  "That puts" -> "That will put" ; "we can" -> "we could". "Also brings" -> "will also bring".